pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -21,115 +21,472 @@ import (
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+	/*
+		- Equivalence Class Testing
+			- case 1.1: success to create index with 1 uncommitted insert index
+			- case 1.2: success to create index with 100 uncommitted insert index
+			- case 2.1: success to create index with 1 uncommitted delete index
+			- case 2.2: success to create index with 100 uncommitted delete index
+			- case 3.1: success to create index with 1 uncommitted insert & delete index
+			- case 3.2: success to create index with 100 uncommitted insert & delete index
+
+		- Boundary Value Testing
+			- case 1.1: success to create index with 0 uncommitted index
+			- case 2.1: success to create index with invalid dimension
+
+		- Decision Table Testing
+			// with uncommitted index count 100
+			- case 1.1: success to create index with poolSize > uncommitted index count
+			- case 1.2: success to create index with poolSize < uncommitted index count
+			- case 1.3: success to create index with poolSize = uncommitted index count
+			- case 1.4: success to create index with poolSize = 0
+	*/
+	tests := []test{
+		{
+			name: "Equivalence Class Testing case 1.1: success to create index with 1 uncommitted insert index",
+			args: args{
+				c: &payload.Control_CreateIndexRequest{
+					PoolSize: 1,
+				},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, a args, s *server) {
+				ctx := context.Background()
+				if err := genAndInsertReq(ctx, s, 1); err != nil {
+					t.Fatal(err)
+				}
+			},
+			want: want{
+				wantRes: &payload.Empty{},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 1.2: success to create index with 100 uncommitted insert index",
+			args: args{
+				c: &payload.Control_CreateIndexRequest{
+					PoolSize: 100,
+				},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, a args, s *server) {
+				ctx := context.Background()
+				if err := genAndInsertReq(ctx, s, 100); err != nil {
+					t.Fatal(err)
+				}
+			},
+			want: want{
+				wantRes: &payload.Empty{},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 2.1: success to create index with 1 uncommitted delete index",
+			args: args{
+				c: &payload.Control_CreateIndexRequest{
+					PoolSize: 1,
+				},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, a args, s *server) {
+				ctx := context.Background()
+				if err := genAndInsertReq(ctx, s, 1); err != nil {
+					t.Fatal(err)
+				}
+				if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+					PoolSize: 1,
+				}); err != nil {
+					t.Fatal(err)
+				}
+				if err := genAndRemoveReq(ctx, s, 1); err != nil {
+					t.Fatal(err)
+				}
+			},
```

Comment:
It is better to use restore function with `initNGT`, then send remove request

Reply:
@vankichi 
to make the test case simple and make the test case visible (we can see that ngt have 1 index inserted before remove in test implementation, but with `initNGT` we cannot), I think the current implementation is better.
what do you think?